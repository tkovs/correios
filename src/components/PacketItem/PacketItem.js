import React from 'react'
import { View, StyleSheet, TouchableHighlight } from 'react-native'
import { ActivityIndicator, Badge, Text } from 'react-native-paper'
import PropTypes from 'prop-types'
import moment from 'moment'
import first from 'lodash/first'
import last from 'lodash/last'
import find from 'lodash/find'
import isNil from 'lodash/isNil'

import { colors } from '../../styles/theme'

const getPassedDays = (packetStatus, firstShippingUpdate, lastShippingUpdate) =>
  packetStatus === 'delivered'
    ? lastShippingUpdate.diff(firstShippingUpdate, 'days')
    : moment().diff(firstShippingUpdate, 'days')

const PacketItem = ({ onClick, packet, statusList }) => {
  const { code } = packet
  const activityIndicatorSize = 12
  const status = find(statusList, { code }) || {}
  const firstShippingUpdate = moment(last(packet.statuses).datetime)
  const lastShippingUpdate = moment(first(packet.statuses).datetime)
  const formattedLastUpdate = moment(lastShippingUpdate).format('D MMM')
  const passedDays = getPassedDays(
    packet.status,
    firstShippingUpdate,
    lastShippingUpdate
  )
  const lastView = moment(packet.lastView)
  const lastUpdate = moment(packet.lastUpdate)

  const isViewed = lastView.isAfter(lastUpdate)

  return (
    <TouchableHighlight underlayColor="#E5E5E5" onPress={onClick}>
      <View style={styles.container}>
        <View style={styles.left}>
          <View style={styles.titleContainer}>
            <Text style={styles.title}>
              {packet.title} {!isViewed && '(n√£o visualizado)'}
            </Text>
            <ActivityIndicator
              animating={!isNil(status.pending)}
              size={activityIndicatorSize}
              color={colors.blue}
            />
          </View>
          <Text
            style={
              packet.status === 'delivered'
                ? styles.statusDelivered
                : styles.statusPending
            }
          >
            {packet.status}
          </Text>
          <Text style={styles.code}>{code}</Text>
        </View>
        <View style={styles.right}>
          <Badge>{packet.mode}</Badge>
          <Text>{formattedLastUpdate}</Text>
          <Text>{`${passedDays} days`}</Text>
        </View>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 8,
    color: colors.snow,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  right: {
    alignItems: 'flex-end',
  },
  title: {
    color: colors.licorice,
    marginRight: 8,
  },
  titleContainer: {
    flexDirection: 'row',
  },
  statusDelivered: {
    color: colors.green,
    fontWeight: 'bold',
  },
  statusPending: {
    color: colors.red,
    fontWeight: 'bold',
  },
  code: {
    color: colors.gray,
  },
})

PacketItem.defaultProps = {
  onClick: null,
  statusList: [],
}

PacketItem.propTypes = {
  packet: PropTypes.shape({
    title: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired,
    mode: PropTypes.string.isRequired,
    lastUpdate: PropTypes.objectOf(Date).isRequired,
    lastView: PropTypes.objectOf(Date).isRequired,
    statuses: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
  onClick: PropTypes.func,
  statusList: PropTypes.arrayOf(PropTypes.object),
}

export default PacketItem
