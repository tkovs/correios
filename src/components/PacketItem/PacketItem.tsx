import React, { FC } from 'react'
import { View, StyleSheet, TouchableHighlight } from 'react-native'
import { ActivityIndicator, Badge, Text } from 'react-native-paper'
import moment, { Moment } from 'moment'
import first from 'lodash/first'
import last from 'lodash/last'
import find from 'lodash/find'
import isNil from 'lodash/isNil'

import { colors } from '../../styles/theme'

const getPassedDays: (
  packetStatus: string,
  firstShippingUpdate: Moment,
  lastShippingUpdate: Moment
) => number = (packetStatus, firstShippingUpdate, lastShippingUpdate) =>
  packetStatus === 'delivered'
    ? lastShippingUpdate.diff(firstShippingUpdate, 'days')
    : moment().diff(firstShippingUpdate, 'days')

interface PacketItemProps {
  packet: Packet
  onClick: (any) => void
  statusList: StatusListItem[]
}

type Props = PacketItemProps

const PacketItem: FC<Props> = ({ onClick, packet, statusList }: Props) => {
  const { code } = packet
  const activityIndicatorSize = 12
  const status = find(statusList, { code }) || {}
  const firstShippingUpdate = moment(last(packet.statuses).datetime)
  const lastShippingUpdate = moment(first(packet.statuses).datetime)
  const formattedLastShippingUpdate = moment(lastShippingUpdate).format('D MMM')
  const passedDays = getPassedDays(
    packet.status,
    firstShippingUpdate,
    lastShippingUpdate
  )
  const lastView = moment(packet.lastView)
  const updatedAt = moment(packet.updatedAt)

  const isViewed = !isNil(packet.lastView) && lastView.isAfter(updatedAt)

  return (
    <TouchableHighlight underlayColor="#E5E5E5" onPress={onClick}>
      <View style={styles.container}>
        <View>
          <View style={styles.titleContainer}>
            <Text accessibilityStates={['disabled']} style={styles.title}>
              {packet.title} {!isViewed && '(n√£o visualizado)'}
            </Text>
            <ActivityIndicator
              accessibilityStates={['disabled']}
              animating={!isNil(status.pending)}
              size={activityIndicatorSize}
              color={colors.blue}
            />
          </View>
          <Text
            accessibilityStates={['disabled']}
            style={
              packet.status === 'delivered'
                ? styles.statusDelivered
                : styles.statusPending
            }
          >
            {packet.status}
          </Text>
          <Text accessibilityStates={['disabled']} style={styles.code}>
            {code}
          </Text>
        </View>
        <View style={styles.right}>
          <Badge>{packet.mode}</Badge>
          <Text accessibilityStates={['disabled']}>
            {formattedLastShippingUpdate}
          </Text>
          <Text accessibilityStates={['disabled']}>{`${passedDays} days`}</Text>
        </View>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 8,
    color: colors.snow,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  right: {
    alignItems: 'flex-end',
  },
  title: {
    color: colors.licorice,
    marginRight: 8,
  },
  titleContainer: {
    flexDirection: 'row',
  },
  statusDelivered: {
    color: colors.green,
    fontWeight: 'bold',
  },
  statusPending: {
    color: colors.red,
    fontWeight: 'bold',
  },
  code: {
    color: colors.gray,
  },
})

PacketItem.defaultProps = {
  statusList: [],
}

export default PacketItem
