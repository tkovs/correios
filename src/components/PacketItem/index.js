import React from 'react'
import PropTypes from 'prop-types'
import { useNavigation } from '@react-navigation/native'

import PacketItem from './PacketItem'

function PacketItemContainer(props) {
  const { packet } = props
  const navigation = useNavigation()

  const navigateToPacketItemInfo = () =>
    navigation.navigate('PacketInfo', { code: packet.code })

  return <PacketItem onClick={navigateToPacketItemInfo} {...props} />
}

PacketItemContainer.propTypes = {
  packet: PropTypes.shape({
    title: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    code: PropTypes.string.isRequired,
    mode: PropTypes.string.isRequired,
    statuses: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
}

export default PacketItemContainer
