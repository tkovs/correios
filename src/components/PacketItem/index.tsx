import React, { FC } from 'react'
import { connect, MapStateToProps } from 'react-redux'
import { useNavigation } from '@react-navigation/native'

import PacketItem from './PacketItem'
import { statusListSelector } from '../../store/selectors/packets'

interface StateProps {
  statusList: StatusListItem[]
}

interface OwnProps {
  packet: Packet
}

type Props = StateProps & OwnProps

const PacketItemContainer: FC<Props> = (props: Props) => {
  const { packet, statusList } = props
  const navigation = useNavigation()

  const navigateToPacketItemInfo: () => void = () =>
    navigation.navigate('PacketInfo', { code: packet.code })

  return (
    <PacketItem
      onClick={navigateToPacketItemInfo}
      packet={packet}
      statusList={statusList}
    />
  )
}

const mapStateToProps: MapStateToProps<StateProps, OwnProps> = (state) => ({
  statusList: statusListSelector(state),
})

PacketItemContainer.defaultProps = {
  statusList: [],
}

export default connect(mapStateToProps)(PacketItemContainer)
