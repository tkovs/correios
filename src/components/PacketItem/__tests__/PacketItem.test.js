import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import PacketItem from '../PacketItem'

// TODO: Migrate test to redux

const mockStatuses = [
  {
    status: 'Objeto entregue ao destinatÃ¡rio',
    datetime: new Date('2020-01-05T00:01:01'),
    locale: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
  {
    status: 'Objeto postado',
    datetime: new Date('2020-01-01T00:01:01'),
    locale: {
      place: 'CEE CENTRO',
      city: 'Rio De Janeiro',
      state: 'RJ',
    },
  },
]

describe('PacketItem', () => {
  describe('with full packet', () => {
    it('renders correctly', () => {
      const mockPacket = {
        title: 'Apple Pencil',
        status: 'delivered',
        code: 'PW086958115BR',
        mode: 'SEDEX',
        statuses: mockStatuses,
        lastView: new Date(2020, 1, 7),
        lastUpdate: new Date(2020, 1, 6),
      }
      const { baseElement } = render(<PacketItem packet={mockPacket} />)
      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('with not viewed packet', () => {
    it('renders correctly', () => {
      const mockPacket = {
        title: 'Apple Pencil',
        status: 'delivered',
        code: 'PW086958115BR',
        mode: 'SEDEX',
        statuses: mockStatuses,
        lastView: new Date(2020, 1, 5),
        lastUpdate: new Date(2020, 1, 6),
      }
      const { baseElement } = render(<PacketItem packet={mockPacket} />)
      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('with updating packet', () => {
    it('renders correctly', () => {
      const mockPacket = {
        title: 'Apple Pencil',
        status: 'delivered',
        code: 'PW086958115BR',
        mode: 'SEDEX',
        statuses: mockStatuses,
        lastView: new Date(2020, 1, 7),
        lastUpdate: new Date(2020, 1, 6),
      }
      const mockStatusList = [
        {
          code: 'PW086958115BR',
          pending: true,
          error: false,
        },
      ]
      const { baseElement } = render(
        <PacketItem packet={mockPacket} statusList={mockStatusList} />
      )
      expect(baseElement).toMatchSnapshot()
    })
  })
})
