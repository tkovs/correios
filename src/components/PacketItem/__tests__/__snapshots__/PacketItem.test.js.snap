// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacketItem with full packet renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={null}
      underlayColor="#E5E5E5"
    >
      <View
        style={
          Object {
            "color": "#f9f9f9",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#141414",
                    "marginRight": 8,
                  },
                ]
              }
            >
              Apple Pencil
               
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 12,
                    "opacity": 0,
                    "width": 12,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": 0,
                              },
                              Object {
                                "rotate": "-165deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "top": 6,
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": -6,
                              },
                              Object {
                                "rotate": "345deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#3E8989",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            delivered
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#939393",
                },
              ]
            }
          >
            PW086958115BR
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#f50057",
                "borderRadius": 10,
                "color": "#ffffff",
                "fontFamily": "System",
                "fontSize": 10,
                "fontWeight": "400",
                "height": 20,
                "lineHeight": 20,
                "minWidth": 20,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 4,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            SEDEX
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            5 Jan
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            4 days
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  </View>
</View>
`;

exports[`PacketItem with not viewed packet renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={null}
      underlayColor="#E5E5E5"
    >
      <View
        style={
          Object {
            "color": "#f9f9f9",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#141414",
                    "marginRight": 8,
                  },
                ]
              }
            >
              Apple Pencil
               
              (não visualizado)
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 12,
                    "opacity": 0,
                    "width": 12,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": 0,
                              },
                              Object {
                                "rotate": "-165deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "top": 6,
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": -6,
                              },
                              Object {
                                "rotate": "345deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#3E8989",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            delivered
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#939393",
                },
              ]
            }
          >
            PW086958115BR
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#f50057",
                "borderRadius": 10,
                "color": "#ffffff",
                "fontFamily": "System",
                "fontSize": 10,
                "fontWeight": "400",
                "height": 20,
                "lineHeight": 20,
                "minWidth": 20,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 4,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            SEDEX
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            5 Jan
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            4 days
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  </View>
</View>
`;

exports[`PacketItem with updating packet renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={null}
      underlayColor="#E5E5E5"
    >
      <View
        style={
          Object {
            "color": "#f9f9f9",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#141414",
                    "marginRight": 8,
                  },
                ]
              }
            >
              Apple Pencil
               
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 12,
                    "opacity": 1,
                    "width": 12,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": 0,
                              },
                              Object {
                                "rotate": "-165deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": 12,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 6,
                          "overflow": "hidden",
                          "top": 6,
                          "width": 12,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 12,
                            "transform": Array [
                              Object {
                                "translateY": -6,
                              },
                              Object {
                                "rotate": "345deg",
                              },
                            ],
                            "width": 12,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "borderColor": "#01417F",
                                "borderRadius": 6,
                                "borderWidth": 1.2,
                                "height": 12,
                                "width": 12,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#3E8989",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            delivered
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#939393",
                },
              ]
            }
          >
            PW086958115BR
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#f50057",
                "borderRadius": 10,
                "color": "#ffffff",
                "fontFamily": "System",
                "fontSize": 10,
                "fontWeight": "400",
                "height": 20,
                "lineHeight": 20,
                "minWidth": 20,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 4,
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            SEDEX
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            5 Jan
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            4 days
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  </View>
</View>
`;
